plugins/source/aws/resources/services/config/config_rule_compliance_details.go

```diff
@@ -0,0 +1,65 @@
+package config
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/service/configservice"
+	"github.com/aws/aws-sdk-go-v2/service/configservice/types"
+	"github.com/cloudquery/cloudquery/plugins/source/aws/client"
+	"github.com/cloudquery/plugin-sdk/v2/schema"
+	"github.com/cloudquery/plugin-sdk/v2/transformers"
+)
+
+func configRuleComplianceDetails() *schema.Table {
+	tableName := "aws_config_config_rule_compliance_details"
+	return &schema.Table{
+		Name:        tableName,
+		Description: `https://docs.aws.amazon.com/config/latest/APIReference/API_GetComplianceDetailsByConfigRule.html`,
+		Resolver:    fetchConfigConfigRuleComplianceDetails,
+		Multiplex:   client.ServiceAccountRegionMultiplexer(tableName, "config"),
+		// no primary key because all the relevant candidate fields can either be null or are not
+		// uniquely identifying of a resource. For example, ResourceEvaluationId can be null,
+		// and so can ResultToken. However, hashing the entire object can work because a combination of
+		// all fields must be unique.
+		Transform: transformers.TransformWithStruct(&types.EvaluationResult{}),
+		Columns: []schema.Column{
+			client.DefaultAccountIDColumn(true),
+			client.DefaultRegionColumn(true),
+			{
+				Name:     "config_rule_name",
+				Type:     schema.TypeString,
+				Resolver: schema.ParentColumnResolver("config_rule_name"),
+			},
+			{
+				Name:     "_evaluation_hash",
```

Comment:
I'm not sure this is the best way to go; we could also not have a PK at all and have it default to `_cq_id`. Thoughts on this?

Reply:
Defaulting to `_cq_id` could be better because if the user wants deterministic they can set `deterministic_cq_id` in the spec and have the _cq_id generated from all column values automatically. On the other hand that setting affects all tables.

Reply:
Yeah exactly, `deterministic_cq_id` affects all tables which is probably not what users would want. We do use this `ResolveObjectHash` technique in two other places as well, but I think that predated `deterministic_cq_id`. What do you think - just use `_cq_id`?

Reply:
I'm unsure but yes the `ResolveObjectHash` method predates deterministic_cq_id so it's probably better to just keep it simple.